### PR TITLE
Improve the C# style guide and mention C# 7.0 as minimum version

### LINKED ISSUE #2540: "Mouse and input coordinates" example doesn't follow the C# style guide
In the [Mouse and input coordinates](https://docs.godotengine.org/en/latest/tutorials/inputs/mouse_and_input_coordinates.html) C# example snippet, C# 7.0's operator pattern matching is used, but the [C# style guide](https://github.com/godotengine/godot-docs/blob/master/getting_started/scripting/c_sharp/c_sharp_style_guide.rst) advises to stick to C# 6.0 for now.

*Originally reported here: https://godotengine.org/qa/46714/does-godot-mono-support-operator-pattern-matching-casting*

### PR DESCRIPTION
**Pull request by Calinou**
This closes #2540, while also improving formatting and removing some redundant information.

### CODE REVIEW

File: getting_started/scripting/c_sharp/c_sharp_style_guide.rst [hunk 1]
```diff
@@ -298,9 +295,10 @@ Other considerations
 
  * Use explicit access modifiers.
  * Use properties instead of non-private fields.
- * Use modifiers in this order: *'public/protected/private/internal virtual/override/abstract/new static readonly'*.
- * Avoid using fully qualified names or *'this.'* prefix for members when it's not necessary.
- * Remove unused *'using'* statements and unnecessary parentheses.
- * Consider omitting default initial value for a type.
+ * Use modifiers in this order:
+   ``public``/``protected``/``private``/``internal virtual``/``override``/``abstract``/``new static readonly``.
```
Comment (neikeq) [inline getting_started/scripting/c_sharp/c_sharp_style_guide.rst]: The last `new static readonly` should be 3 separate modifiers.

Comment (Calinou) [reply getting_started/scripting/c_sharp/c_sharp_style_guide.rst]: Done. Should `internal virtual` be separated as well?

Comment (neikeq) [reply getting_started/scripting/c_sharp/c_sharp_style_guide.rst]: Yes, didn't notice that one.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**